src/Core/x/xGroup.h

```diff
@@ -7,6 +7,7 @@ struct xGroupAsset : xBaseAsset
 {
     uint16 itemCount;
     uint16 groupFlags;
+    uint32* link;
```

Comment:
`link` should not be added here, it's intentional that it's not included in the original struct. Usually the way xLinkAsset* works is it points to the end of the asset struct, like this:

```
(xLinkAsset*)(asset + 1)
```

I see you did it like this in `xGroupInit`:

```
(xLinkAsset*)(&asset->link)
```

which, yes, does technically accomplish the same thing, but it's not how it was originally done. It might also be the reason behind your non-match.